python/tvm/ir/memory_pools.py

```diff
@@ -245,7 +245,7 @@ def __init__(
         )
 
 
-@register_object("ir.ConstantMemoryPools")
```

Comment:
A bit off the topic, but to some extent, I believe it might be less ideal if we expose those feature-dependent component in `tvm/ir` folder, which is supposed to be generic.

Reply:
Yeah, I wasn't sure how much refactoring to include in this PR.  There were some clear inconsistencies, such as the wrong object name being in `@register_object`, but this feels like something that should be in the `tir.usmp` namespace.

Reply:
Agreed. This seems less organized. Let's leave this to future change